vmm/src/igvm/igvm_loader.rs

```diff
@@ -64,6 +73,41 @@ enum ParameterAreaState {
     Inserted,
 }
 
+#[cfg(all(feature = "sev_snp", target_arch = "x86_64"))]
+fn generate_memory_map(
+    guest_mem: &GuestMemoryMmap,
+) -> Result<Vec<IGVM_VHS_MEMORY_MAP_ENTRY>, Error> {
+    let mut memory_map = Vec::new();
+    let ram_regions = arch::generate_ram_regions(guest_mem);
+    arch::validate_ram_regions(&ram_regions).map_err(Error::InvalidGuestMemmap)?;
+
+    // Create the memory map entry for memory region before the gap
+    let first_region_end = ram_regions.first().unwrap().1;
+    let high_ram_start = arch::layout::HIGH_RAM_START.raw_value();
+    memory_map.push(IGVM_VHS_MEMORY_MAP_ENTRY {
```

Comment:
It would be wise to define some macro or from() definition to convert this to avoid code duplication.

Reply:
I cannot impl From trait for IGVM_VHS_MEMORY_MAP_ENTRY because it is defined outside this crate. So, I implemented a function instead for the same.